src/app/public/modules/restricted-view/restricted-view-auth.service.spec.ts

```diff
@@ -0,0 +1,75 @@
+import {
+  SkyRestrictedViewAuthService
+} from './restricted-view-auth.service';
+
+describe('SkyRestrictedViewAuthService', () => {
+  let authService: SkyRestrictedViewAuthService;
+  let mockAuthTokenProvider: any;
+
+  it('should authenticate when JWT contains 1bb.perms', () => {
+    mockAuthTokenProvider = {
+      getToken(args: any) {
+        return Promise.resolve(true);
+      }
+    };
+    authService = new SkyRestrictedViewAuthService(mockAuthTokenProvider as any);
+    spyOn(authService.isAuthenticated, 'getValue').and.callThrough();
+    spyOn(authService['auth'], 'getToken').and.returnValue(Promise.resolve('')).and.callThrough();
+    spyOn<any>(authService, 'decode').and.returnValue({
+      '1bb.perms': 1
+    });
+    authService.isAuthenticated.subscribe(next => {
+      if (next) {
```

Comment:
I'm seeing a few `if (next) {` in your unit tests. This can be problematic because if `next` is undefined (or false), the test will still pass. If you must have the condition in here, I would put a `fail` message in the `else` to make sure the unit test is actually passing.

```
if (next) {
  expect(...);
} else {
  fail('It should fail!');
}
```

Reply:
It might also be a good idea to use jasmine's `done` callback since these are asynchronous tests.